### PR TITLE
Uvs 49: Implement tag mappings API

### PR DESCRIPTION
**Pull request by VolodymyrFihurniak**

### CODE REVIEW

File: server/src/dto/taggedFileDTO.ts [hunk 1]
```diff
@@ -60,35 +60,76 @@ class ShortTagsDTO {
   };
 }
 
+class FileMappingDTO {
```
Comment (denbykov) [inline server/src/dto/taggedFileDTO.ts:63]: I believe it should be named TagMappingDTO

File: server/src/dto/taggedFileDTO.ts [hunk 2]
```diff
@@ -60,35 +60,76 @@ class ShortTagsDTO {
...
+      json.file_mapping_artist,
+      json.file_mapping_album,
+      json.file_mapping_year,
+      json.file_mapping_track_number,
+      json.file_mapping_picture
+    );
+  };
+}
+
 class TaggedFileDTO {
```
Comment (denbykov) [inline server/src/dto/taggedFileDTO.ts:99]: Do not change this DTO

File: server/src/dto/taggedFileDTO.ts (hunk 1 above)
Comment (denbykov) [reply server/src/dto/taggedFileDTO.ts:63]: And moved to the separate file

File: server/src/entities/file.ts [hunk 3]
```diff
@@ -25,24 +25,54 @@ export class ShortTags {
   }
 }
 
-export class File {
+class FileMapping {
```
Comment (denbykov) [inline server/src/entities/file.ts:28]: Rename to TagMapping and move to the separate file

File: server/src/entities/file.ts [hunk 4]
```diff
@@ -25,24 +25,54 @@ export class ShortTags {
...
+    this.title = title;
+    this.artist = artist;
+    this.album = album;
+    this.year = year;
+    this.trackNumber = trackNumber;
+    this.pictureId = pictureId;
+  }
+}
+
+class File {
```
Comment (denbykov) [inline server/src/entities/file.ts:53]: Do not change File Entity. Create another one, that will include both file and mapping.

**Review by denbykov — CHANGES_REQUESTED** [review]

File: server/src/controllers/tagMappingController.ts [hunk 5]
```diff
@@ -79,7 +79,30 @@ class TagMappingController extends BaseController {
     next: Express.NextFunction
   ): Promise<Express.Response | void> => {
     try {
-      const fileId = Number(request.params.fileId);
+      const fileId = request.params.fileId;
+
+      if (Object.keys(request.body).length == 0) {
```
Comment (VolodymyrFihurniak) [inline server/src/controllers/tagMappingController.ts:84]: This code can be written as follows:
```
const requiredFields = ['title', 'artist', 'album', 'picture', 'year', 'trackNumber'];

for (let field of requiredFields) {
  if (!request.body[field]) {
    return response.status(400).json({ error: `${field.charAt(0).toUpperCase() + field.slice(1)} is required` });
  }
}
```

**Comment by VolodymyrFihurniak** [comment]
Use method toString() conversion from number to string type

**Review by denbykov — APPROVED** [review]

**Pull request merged**